### PR TITLE
[ML] Stream Anthropic Completion

### PR DESCRIPTION
**Pull request by prwhelan**
Enable chat completion streaming responses for Anthropic's server sent events.

### CODE REVIEW

**Comment by prwhelan** [comment]
@elasticmachine update branch

File: x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/anthropic/AnthropicStreamingProcessorTests.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    public void testParseSuccess() {
+        var item = events("""
+            {
+                "type": "message_start",
+                "message": {
+                    "id": "a cool id",
+                    "type": "message",
+                    "role": "assistant",
+                    "content": [],
+                    "model": "claude, probably",
```
Comment (davidkyle) [inline x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/anthropic/AnthropicStreamingProcessorTests.java:48]: 😆

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by prwhelan** [comment]
@elasticmachine update branch

**Comment by prwhelan** [comment]
run elasticsearch-ci/packaging-tests-windows-sample

**Comment by prwhelan** [comment]
@elasticmachine update branch

**Pull request merged**